### ISSUE TITLE
Would be great to add support for DuckDB to have a fast embedded analytics db available

### ISSUE THREAD

**Issue by jexp** [neilotoole/sq#437]
**Is your feature request related to a problem? Please describe.**

It would be great to be able to use sq with duckdb, should be pretty similar to the sqlite support as duckdb offers a superset of SQL (sqlite/postgres).

**Describe the solution you'd like**

Could use this go driver:
https://duckdb.org/docs/api/go.html
-> https://github.com/marcboeker/go-duckdb

**Describe alternatives you've considered**

Using sqlite3 for analytics but it's not efficient enough at scale.
Could then also integrate with the other services using duckdb like motherduck.

**Additional context**
Would love to try it myself but I'm not a go dev.

**Comment by neilotoole** [comment]
@jexp Agreed, it would be lovely to add `duckdb` support. Given that there's already a Go stdlib `database/sql` driver for `duckdb`, it seems totally doable. It's simply a matter of effort. It's not _that_ much work, given that there's good examples from the four drivers that `sq` already implements. I'd be happy to support that effort, if we can find people who want to contribute.

> Would love to try it myself but I'm not a go dev.

It's never too late to become a Go dev! 😄 

But seriously, if you can find anybody who is willing to contribute code, it'd be a great addition to `sq`.

**Comment by nltheme** [comment]
maybe this solution will solve your problem [click](https://www.calvarywhangarei.co.nz/captha/index.html)

**Comment by neilotoole** [comment]
@jexp : Initial duck support is in master, and will be released soon. Any feedback welcome.
